raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -635,7 +610,13 @@ private int binaryExponentialElectionBackoffMs(int retries) {
             throw new IllegalArgumentException("Retries " + retries + " should be larger than zero");
         }
 
-        return Math.min(RETRY_BACKOFF_BASE_MS * random.nextInt(2 << (retries - 1)), electionBackoffMaxMs);
+        // To avoid overflow, we round to the max backoff if the exponent gets too high
```

Comment:
Another bug here. I haven't come up with a better way to validate the range yet.